near-bindgen-core/src/code_generator/impl_item_method_info.rs

```diff
@@ -52,8 +52,20 @@ impl ImplItemMethodInfo {
             returns,
             result_serializer,
             is_init,
+            is_payable,
             ..
         } = attr_signature_info;
+        let deposit_check = if *is_payable {
+            // No check if method is payable
+            quote! {}
+        } else {
+            // If method is not payable, do a check to make sure that it doesn't consume deposit
+            quote! {
+                if near_bindgen::env::attached_deposit() != 0 {
+                    panic!("Method doesn't accept deposit");
```

Comment:
I think we should panic with env instead in case we switch to no_std in the future, so we don't need to capture inner panic thought the hook.
```suggestion
                    near_bindgen::env::panic(b"Method doesn't accept deposit");
```

Reply:
there is a https://doc.rust-lang.org/nomicon/panic-handler.html

Actually it's just generated some user-land code. Should we recommend users to not use `panic!`? 

Btw, there is a #[no_std] alternative https://doc.rust-lang.org/core/panic/struct.PanicInfo.html 

But I actually ok with both versions.

Reply:
It's probably faster to panic with `env` than with macro.

Reply:
third-party will never use the `near_bindgen::env::panic`, they are using a standard functions. I'm fine with `near_bindgen::env::panic` here, but slightly into to use a standard interfaces and adopt them as a default strategy.